### ISSUE TITLE
\verb*| | and \textvisiblespace produces different visible space symbols, with pdfLaTeX

### ISSUE THREAD

**Issue by muzimuzhi** [latex3/latex2e#70]
## Brief outline of the bug
With pdfLaTeX, `\verb*| |` and `\textvisiblespace` produces visible space symbols with different appearances.

The issue may relates to [latex2e/issues/69: \verb* cannot produce visible space with XeLaTeX](https://github.com/latex3/latex2e/issues/69), but I'm not sure.

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}
\documentclass{minimal}
\begin{document}
\verb*|x x|
\ttfamily\textvisiblespace
\end{document}
```

## Log file (required) and possibly PDF file  
[verbatim.log](https://github.com/latex3/latex2e/files/2342741/verbatim.log)
[verbatim.pdf](https://github.com/latex3/latex2e/files/2342742/verbatim.pdf)

**Comment by davidcarlisle** [comment]
yes thanks, the fact that they are different was one reason that the earlier referenced issue wasn't fixed straight away as this it complicates the issue.

In (pdf)latex `\verb*` uses character 32 (space) from the tt font which (for computer modern at least) has a visible space character rather than a space in that slot.  `\textvisiblespace` is encoding-dependent but by default (and in almost all cases) it uses a shallower symbol that is not a font character at all but three tex rules drawing a ␣.    fontspec currently makes both use U+2423 (␣) but not all fonts have a usable character in that slot and as you see the visible appearance of the three forms can be quite different.

So we will do something to fix both these issues, but we are discussing what is the best (and most compatible) change to make.
